canonical-web-and-design/vanilla-framework
# Style range input as slider by default

## Done

Slider pattern styles are now applied as default styling for `<input type=range>` form inputs.
Slider pattern itself is still available to allow version with built in input.

Documentation was updated with new base forms example, default slider example was removed.

### NOTE
This is potentially breaking change.
If any project used to have unstyled (or custom styled `<input type=range>` this input will now get Vanilla slider styling.

I made a search around our project to find uses of range input and haven't found anything suspicious. Any clear occurrences where already using `p-slider`.

While `p-slider` class itself is not needed anymore (and is removed from Vanilla) existing uses of it will still work as before because they were required to be used on range input.

## QA

Note: CodePen examples in docs pages use published 2.5.0 version of Vanilla, so will not show the changes.
To see changes look at component example pages directly.

- Pull code
- Run `./run serve --watch`
- Open http://0.0.0.0:8101/ or [demo](https://vanilla-framework-canonical-web-and-design-pr-2730.run.demo.haus/)
- Go to forms docs, see [info about range input](https://vanilla-framework-canonical-web-and-design-pr-2730.run.demo.haus/base/forms/#range) (note that codepen will have styling from previous release)
- Go to [slider page](https://vanilla-framework-canonical-web-and-design-pr-2730.run.demo.haus/patterns/slider/), see it updated
- Go to [forms input range example](https://vanilla-framework-canonical-web-and-design-pr-2730.run.demo.haus/examples/base/forms/range/), check if it works (same as pattern before)

## Details

Fixes #2602
